File: Dddml.Wms.HttpServices.PhpClient/Wms/Domain/AttributeSetInstanceExtensionFieldMvoStateEventId.php
<?php

namespace Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\Serializer\Type\Long;
use Dddml\StringIdInterface;

class AttributeSetInstanceExtensionFieldMvoStateEventId implements StringIdInterface
{
    /**
     * @Type("Wms\Domain\AttributeSetInstanceExtensionFieldId")
     */
    private $attributeSetInstanceExtensionFieldId;

    /**
     * @return AttributeSetInstanceExtensionFieldId
     */
    public function getAttributeSetInstanceExtensionFieldId()
    {
        if(!$this->attributeSetInstanceExtensionFieldId) {
            $this->attributeSetInstanceExtensionFieldId = new AttributeSetInstanceExtensionFieldId(); 
        }
        return $this->attributeSetInstanceExtensionFieldId;
    }

    /**
     * @param AttributeSetInstanceExtensionFieldId $attributeSetInstanceExtensionFieldId
     */
    public function setAttributeSetInstanceExtensionFieldId($attributeSetInstanceExtensionFieldId)
    {
        $this->attributeSetInstanceExtensionFieldId = $attributeSetInstanceExtensionFieldId;
    }

    /**
     * @Type("Dddml\Serializer\Type\Long")
     */
    private $attrSetInstEFGroupVersion;

    /**
     * @return Long
     */
    public function getAttrSetInstEFGroupVersion()
    {
        return $this->attrSetInstEFGroupVersion;
    }

    /**
     * @param Long $attrSetInstEFGroupVersion
     */
    public function setAttrSetInstEFGroupVersion($attrSetInstEFGroupVersion)
    {
        $this->attrSetInstEFGroupVersion = $attrSetInstEFGroupVersion;
    }



    /**
     * @var AttributeSetInstanceExtensionFieldMvoStateEventIdFlattenedDto
     */
    private $idFlattenedDto;

    /**
     * @return string
     */
    public function toString()
    {
        if (!$this->idFlattenedDto) {
            $this->idFlattenedDto = new AttributeSetInstanceExtensionFieldMvoStateEventIdFlattenedDto($this);
        }

        return $this->idFlattenedDto->toString();
    }

    /**
     * @param string $idStr
     *
     * @return AttributeSetInstanceExtensionFieldMvoStateEventId
     */
    public static function createFromString($idStr)
    {
        return (new AttributeSetInstanceExtensionFieldMvoStateEventIdFlattenedDto())
            ->fromString($idStr)
            ->toAttributeSetInstanceExtensionFieldMvoStateEventId();
    }


}

File: Dddml.Wms.JavaCommon/src/org/dddml/wms/domain/AttributeSetInstanceExtensionFieldMvoStateEventId.java
package org.dddml.wms.domain;

import java.io.Serializable;

public class AttributeSetInstanceExtensionFieldMvoStateEventId implements Serializable
{
    private AttributeSetInstanceExtensionFieldId attributeSetInstanceExtensionFieldId = new AttributeSetInstanceExtensionFieldId();

    public AttributeSetInstanceExtensionFieldId getAttributeSetInstanceExtensionFieldId()
    {
        return this.attributeSetInstanceExtensionFieldId;
    }

    public void setAttributeSetInstanceExtensionFieldId(AttributeSetInstanceExtensionFieldId attributeSetInstanceExtensionFieldId)
    {
        this.attributeSetInstanceExtensionFieldId = attributeSetInstanceExtensionFieldId;
    }

    private Long attrSetInstEFGroupVersion;

    public Long getAttrSetInstEFGroupVersion()
    {
        return this.attrSetInstEFGroupVersion;
    }

    public void setAttrSetInstEFGroupVersion(Long attrSetInstEFGroupVersion)
    {
        this.attrSetInstEFGroupVersion = attrSetInstEFGroupVersion;
    }

    public String getAttributeSetInstanceExtensionFieldIdGroupId()
    {
        return getAttributeSetInstanceExtensionFieldId().getGroupId();
    }

    public void setAttributeSetInstanceExtensionFieldIdGroupId(String attributeSetInstanceExtensionFieldIdGroupId)
    {
        getAttributeSetInstanceExtensionFieldId().setGroupId(attributeSetInstanceExtensionFieldIdGroupId);
    }

    public String getAttributeSetInstanceExtensionFieldIdIndex()
    {
        return getAttributeSetInstanceExtensionFieldId().getIndex();
    }

    public void setAttributeSetInstanceExtensionFieldIdIndex(String attributeSetInstanceExtensionFieldIdIndex)
    {
        getAttributeSetInstanceExtensionFieldId().setIndex(attributeSetInstanceExtensionFieldIdIndex);
    }

    AttributeSetInstanceExtensionFieldMvoStateEventId ()
    {
    }

    public AttributeSetInstanceExtensionFieldMvoStateEventId (AttributeSetInstanceExtensionFieldId attributeSetInstanceExtensionFieldId, Long attrSetInstEFGroupVersion)
    {
        this.attributeSetInstanceExtensionFieldId = attributeSetInstanceExtensionFieldId;
        this.attrSetInstEFGroupVersion = attrSetInstEFGroupVersion;
    }

    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        AttributeSetInstanceExtensionFieldMvoStateEventId other = (AttributeSetInstanceExtensionFieldMvoStateEventId)obj;
        return true 
            && (attributeSetInstanceExtensionFieldId == other.attributeSetInstanceExtensionFieldId || (attributeSetInstanceExtensionFieldId != null && attributeSetInstanceExtensionFieldId.equals(other.attributeSetInstanceExtensionFieldId)))
            && (attrSetInstEFGroupVersion == other.attrSetInstEFGroupVersion || (attrSetInstEFGroupVersion != null && attrSetInstEFGroupVersion.equals(other.attrSetInstEFGroupVersion)))
            ;
    }

    @Override
    public int hashCode()
    {
        int hash = 0;
        if (this.attributeSetInstanceExtensionFieldId != null) {
            hash += 13 * this.attributeSetInstanceExtensionFieldId.hashCode();
        }
        if (this.attrSetInstEFGroupVersion != null) {
            hash += 13 * this.attrSetInstEFGroupVersion.hashCode();
        }
        return hash;
    }

}

